src/Libraries/Microsoft.Extensions.AI.Abstractions/Utilities/AIJsonUtilities.Schema.cs

```diff
@@ -351,17 +352,21 @@ static JsonObject ConvertSchemaToObject(ref JsonNode schema)
         }
     }
 
-    private static bool TypeIsArrayContainingInteger(JsonObject schema)
+    private static bool TypeIsIntegerWithStringNumberHandling(JsonSchemaExporterContext ctx, JsonObject schema)
     {
-        if (schema["type"] is JsonArray typeArray)
+        if (ctx.TypeInfo.NumberHandling is not JsonNumberHandling.Strict && schema["type"] is JsonArray typeArray)
```

Comment:
This was a bug found with the additional testing? Yay :)

Reply:
It was a known issue with this mitigation from the start, but yeah, adding a deluge of test cases forced my hand to constrain it to the schemas it was intended to address.